tfx/examples/cifar10/cifar10_pipeline_native_keras.py

```diff
@@ -0,0 +1,183 @@
+                              'metadata.db')
+
+def _create_pipeline(pipeline_name: Text, pipeline_root: Text, data_root: Text,
+                     module_file: Text,
+                     serving_model_dir: Text,
+                     metadata_path: Text,
+                     direct_num_workers: int) -> pipeline.Pipeline:
+  """Implements the cifar10 image classification pipeline using TFX."""
+  input_config = example_gen_pb2.Input(splits=[
+            example_gen_pb2.Input.Split(name='train', pattern='train/*'),
+            example_gen_pb2.Input.Split(name='eval', pattern='test/*')])
+
+  examples = external_input(data_root)
+
+  # Brings data into the pipeline.
+  example_gen = ImportExampleGen(input=examples, input_config=input_config)
+
+  # Computes statistics over data for visualization and example validation.
+  statistics_gen = StatisticsGen(examples=example_gen.outputs['examples'])
+
+  # Generates schema based on statistics files.
+  schema_gen = SchemaGen(
+      statistics=statistics_gen.outputs['statistics'], infer_feature_shape=True)
+
+  # Performs anomaly detection based on statistics and data schema.
+  example_validator = ExampleValidator(
+      statistics=statistics_gen.outputs['statistics'],
+      schema=schema_gen.outputs['schema'])
+
+  # Performs transformations and feature engineering in training and serving.
+  transform = Transform(
+      examples=example_gen.outputs['examples'],
+      schema=schema_gen.outputs['schema'],
+      module_file=module_file)
+
+  def _create_trainer(module_file, instance_name):
+    return Trainer(
+        module_file=module_file,
+        custom_executor_spec=executor_spec.ExecutorClassSpec(GenericExecutor),
+        examples=transform.outputs['transformed_examples'],
+        transform_graph=transform.outputs['transform_graph'],
+        schema=schema_gen.outputs['schema'],
+        train_args=trainer_pb2.TrainArgs(num_steps=10000),
+        eval_args=trainer_pb2.EvalArgs(num_steps=156),
+        instance_name=instance_name)
+
+  # Uses user-provided Python function that trains a Keras model.
+  trainer = _create_trainer(module_file, 'cifar10')
+
+  # Uses TFMA to compute an evaluation statistics over features of a model and
+  # performs quality validation of a candidate model.
+  eval_config = tfma.EvalConfig(
+      model_specs=[tfma.ModelSpec(label_key='label')],
+      slicing_specs=[tfma.SlicingSpec()],
+      metrics_specs=[
+          tfma.MetricsSpec(metrics=[
+              tfma.MetricConfig(
+                  class_name='SparseCategoricalAccuracy',
+                  threshold=tfma.config.MetricThreshold(
+                      value_threshold=tfma.GenericValueThreshold(
+                          lower_bound={'value': 0.8}),
+                      change_threshold=tfma.GenericChangeThreshold(
+                          direction=tfma.MetricDirection.HIGHER_IS_BETTER,
+                          absolute={'value': -1e-10})))
```

Comment:
This seems extremely tight, how about -1e-3 or so? By the way, where did this number come from? Because it seems too tight for most use-cases.

Reply:
Sure we can do that, i copied it from the iris example i remember

Reply:
@1025KB Should we change our examples? This seems to tight for most use-cases.